scripts/stage-release.sh

```diff
@@ -102,6 +105,13 @@ cleanup() {
     fi
 }
 
+setVersion() {
+  local VERSION=$1
+  mvn -q -B versions:set -DnewVersion="${VERSION}" -DgenerateBackupPoms=false -DprocessAllModules=true
+
+  git add '**/pom.xml' 'pom.xml'
+}
```

Comment:
I've just changed this in the main/runtime repo to 

```suggestion
updateVersions() {
  local FROM_VERSION=$1
  local NEW_VERSION=$2
  mvn -q -B versions:set -DnewVersion="${NEW_VERSION}" -DgenerateBackupPoms=false -DprocessAllModules=true

  # Bump version ref in files not controlled by Maven
  # shellcheck disable=SC2046
  ${SED} -i -e "s#${FROM_VERSION//./\\.}#${NEW_VERSION}#g" $(find kubernetes-examples -name "*.yaml" -type f)

  git add '**/*.yaml' '**/pom.xml' 'pom.xml'
}
```

I know we don't need the `sed` command here but I wonder if we should keep API consistency. 

I'm starting to wonder if we should be baking all this scripting into a github action that both projects use?

Reply:
>I'm starting to wonder if we should be baking all this scripting into a github action that both projects use?

+1  I'd been thinking similar things.

I'd been wondering about moving the release machinery to a separate repo and then reusing it from both projects.

I was thinking to keep the mainstay as scripts that you can run locally.  I say this because AFAIK there's no way to run a workflow from a local machine, so that makes testing script slow (involves GitHub). 

But I'm open to alternative ideas.

Reply:
It was an off the cuff thought but I wasn't really think of anything more sophisticated than using the action as the distribution mechanism for the scripts (and standardising the args)